swp/utils/date.py

```diff
@@ -0,0 +1,28 @@
+import re
+import datetime as dt
+
+import datefinder
+
+
+def parse_date(date_str: str, default_date_str: str='1700-01-01') -> dt.date:
```

Comment:
Why did you decide to pass the default as a string when every current usage of this method has a real date to hand? Why the default `1700-01-01`?

I would remove the default all together and return `None` when no date could be parsed. The calling code can decide whether it uses a default or does some other stuff then.